client/post-editor/editor-ground-control/index.jsx

```diff
@@ -294,32 +238,29 @@ export class EditorGroundControl extends PureComponent {
 const mapStateToProps = ( state, ownProps ) => {
 	const siteId = get( ownProps, 'site.ID', null );
 
-	const canUserPublishPosts = canCurrentUser( state, siteId, 'publish_posts' );
-
 	return {
-		canUserPublishPosts,
+		canUserPublishPosts: canCurrentUser( state, siteId, 'publish_posts' ),
 		routeHistory: getRouteHistory( state ),
+		// do not allow publish for unverified e-mails, but allow if the site is VIP
+		userNeedsVerification: ! isCurrentUserEmailVerified( state ) && ! isVipSite( state, siteId ),
 	};
 };
 
-const mapDispatchToProps = dispatch => ( {
-	recordPreviewButtonClick: () =>
-		dispatch(
-			composeAnalytics(
-				recordTracksEvent(
-					`calypso_editor_${ postUtils.isPage( page ) ? 'page' : 'post' }_preview_button_click`
-				),
-				recordGoogleEvent(
-					'Editor',
-					`Clicked Preview ${ postUtils.isPage( page ) ? 'Page' : 'Post' } Button`,
-					`Editor Preview ${ postUtils.isPage( page ) ? 'Page' : 'Post' } Button Clicked`,
-					`editor${ postUtils.isPage( page ) ? 'Page' : 'Post' }ButtonClicked`
-				)
+const mapDispatchToProps = {
+	recordPreviewButtonClick: post => {
+		const postIsPage = isPage( post );
+		return composeAnalytics(
+			recordTracksEvent( `calypso_editor_${ postIsPage ? 'page' : 'post' }_preview_button_click` ),
```

Comment:
I understand this was here before, but as a general note, let's avoid   crafting stat strings like this. It makes it so difficult to find actual usages with search later. For example searching for "calypso_editor_post_preview_button_click" would miss the usage in this file.